mockk-native/src/main/kotlin/io/mockk/impl/instantiation/NativeInstantiator.kt

```diff
@@ -0,0 +1,11 @@
+package io.mockk.impl.instantiation
+
+import kotlin.reflect.KClass
+
+class NativeInstantiator(instanceFactoryRegistry: CommonInstanceFactoryRegistry) :
+    AbstractInstantiator(instanceFactoryRegistry) {
+    @Suppress("UNCHECKED_CAST")
+    override fun <T : Any> instantiate(cls: KClass<T>): T {
```

Comment:
Do you really need this level of reflection? Is it possible to avoid instantiation by KClass, or it's absolutely critical for MockK design?

Reply:
To answer this I need to check usages. One usage would be just object generation for purpose of call signing. This can be replaced by arbitary non-object pointers. Because such object should have only reference equals and identity hashcode. Second usage is returning dummy object for relaxed mocks, so that no NPE is thrown. Such objects can be just zeroed. And more usages I need to check. Of course we can think of some workarounds and framework limitations, but lets focus on proxy/interception first